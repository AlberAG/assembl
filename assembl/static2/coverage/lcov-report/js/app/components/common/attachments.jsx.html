<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for js/app/components/common/attachments.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> / <a href="index.html">js/app/components/common</a> attachments.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">88.89% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>8/9</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">78.57% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>11/14</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">88.89% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>8/9</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">// @flow
import * as React from 'react';
import { Translate } from 'react-redux-i18n';
&nbsp;
type Props = {
  attachments: ?Array&lt;?AttachmentFragment&gt;
};
&nbsp;
const DEFAULT_FILENAME = 'file';
&nbsp;
const Attachments = ({ attachments }: Props) =&gt; (
  &lt;div className="attachments"&gt;
    {attachments &amp;&amp;
      attachments.map((attachment) =&gt; {
        <span class="missing-if-branch" title="if path not taken" >I</span>if (!attachment || !attachment.document) {
<span class="cstat-no" title="statement not covered" >          return null;</span>
        }
        const { externalUrl, mimeType, title } = attachment.document;
        if (mimeType &amp;&amp; mimeType.startsWith('image/')) {
          return null;
        }
&nbsp;
        return (
          &lt;div className="attachment" key={attachment.id}&gt;
            &lt;span className="assembl-icon-synthesis" /&gt;
            &lt;span className="title"&gt;{title || <span class="branch-1 cbranch-no" title="branch not covered" >externalUrl}</span>&lt;/span&gt;
            &lt;a download={title || <span class="branch-1 cbranch-no" title="branch not covered" >DEFAULT_FILENAME}</span> href={externalUrl} type={mimeType} target="_blank" rel="noopener noreferrer"&gt;
              &lt;Translate value="common.attachments.download" /&gt;
            &lt;/a&gt;
          &lt;/div&gt;
        );
      })}
  &lt;/div&gt;
);
&nbsp;
export default Attachments;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Oct 26 2018 15:51:30 GMT+0200 (CEST)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
</body>
</html>
