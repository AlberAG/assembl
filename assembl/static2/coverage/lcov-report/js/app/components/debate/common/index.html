<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for js/app/components/debate/common</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../../index.html">All files</a> js/app/components/debate/common
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">13.93% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>34/244</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0.96% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>2/208</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">8.33% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>7/84</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">14.29% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>34/238</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="announcement.jsx"><a href="announcement.jsx.html">announcement.jsx</a></td>
	<td data-value="16.13" class="pic low"><div class="chart"><div class="cover-fill" style="width: 16%;"></div><div class="cover-empty" style="width:84%;"></div></div></td>
	<td data-value="16.13" class="pct low">16.13%</td>
	<td data-value="31" class="abs low">5/31</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="10" class="abs low">0/10</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="13" class="abs low">0/13</td>
	<td data-value="17.24" class="pct low">17.24%</td>
	<td data-value="29" class="abs low">5/29</td>
	</tr>

<tr>
	<td class="file high" data-value="backButton.jsx"><a href="backButton.jsx.html">backButton.jsx</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	</tr>

<tr>
	<td class="file low" data-value="deletePostButton.jsx"><a href="deletePostButton.jsx.html">deletePostButton.jsx</a></td>
	<td data-value="23.08" class="pic low"><div class="chart"><div class="cover-fill" style="width: 23%;"></div><div class="cover-empty" style="width:77%;"></div></div></td>
	<td data-value="23.08" class="pct low">23.08%</td>
	<td data-value="13" class="abs low">3/13</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="4" class="abs low">1/4</td>
	<td data-value="23.08" class="pct low">23.08%</td>
	<td data-value="13" class="abs low">3/13</td>
	</tr>

<tr>
	<td class="file medium" data-value="deletedPost.jsx"><a href="deletedPost.jsx.html">deletedPost.jsx</a></td>
	<td data-value="50" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 50%;"></div><div class="cover-empty" style="width:50%;"></div></div></td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	</tr>

<tr>
	<td class="file high" data-value="editPostButton.jsx"><a href="editPostButton.jsx.html">editPostButton.jsx</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	</tr>

<tr>
	<td class="file low" data-value="editPostForm.jsx"><a href="editPostForm.jsx.html">editPostForm.jsx</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="41" class="abs low">0/41</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="26" class="abs low">0/26</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="14" class="abs low">0/14</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="41" class="abs low">0/41</td>
	</tr>

<tr>
	<td class="file medium" data-value="headerActions.jsx"><a href="headerActions.jsx.html">headerActions.jsx</a></td>
	<td data-value="75" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 75%;"></div><div class="cover-empty" style="width:25%;"></div></div></td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="4" class="abs medium">3/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="4" class="abs medium">3/4</td>
	</tr>

<tr>
	<td class="file low" data-value="overflowMenu.jsx"><a href="overflowMenu.jsx.html">overflowMenu.jsx</a></td>
	<td data-value="25" class="pic low"><div class="chart"><div class="cover-fill" style="width: 25%;"></div><div class="cover-empty" style="width:75%;"></div></div></td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="4" class="abs low">1/4</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="4" class="abs low">1/4</td>
	</tr>

<tr>
	<td class="file low" data-value="postActions.jsx"><a href="postActions.jsx.html">postActions.jsx</a></td>
	<td data-value="3.85" class="pic low"><div class="chart"><div class="cover-fill" style="width: 3%;"></div><div class="cover-empty" style="width:97%;"></div></div></td>
	<td data-value="3.85" class="pct low">3.85%</td>
	<td data-value="26" class="abs low">1/26</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="30" class="abs low">0/30</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="5" class="abs low">0/5</td>
	<td data-value="4.17" class="pct low">4.17%</td>
	<td data-value="24" class="abs low">1/24</td>
	</tr>

<tr>
	<td class="file high" data-value="replyToCommentButton.jsx"><a href="replyToCommentButton.jsx.html">replyToCommentButton.jsx</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	</tr>

<tr>
	<td class="file high" data-value="sentimentDefinitions.jsx"><a href="sentimentDefinitions.jsx.html">sentimentDefinitions.jsx</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	</tr>

<tr>
	<td class="file low" data-value="sentimentStats.jsx"><a href="sentimentStats.jsx.html">sentimentStats.jsx</a></td>
	<td data-value="25" class="pic low"><div class="chart"><div class="cover-fill" style="width: 25%;"></div><div class="cover-empty" style="width:75%;"></div></div></td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="4" class="abs low">1/4</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="4" class="abs low">1/4</td>
	</tr>

<tr>
	<td class="file low" data-value="sentiments.jsx"><a href="sentiments.jsx.html">sentiments.jsx</a></td>
	<td data-value="6.25" class="pic low"><div class="chart"><div class="cover-fill" style="width: 6%;"></div><div class="cover-empty" style="width:94%;"></div></div></td>
	<td data-value="6.25" class="pct low">6.25%</td>
	<td data-value="16" class="abs low">1/16</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="40" class="abs low">0/40</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="6.25" class="pct low">6.25%</td>
	<td data-value="16" class="abs low">1/16</td>
	</tr>

<tr>
	<td class="file low" data-value="statisticsDoughnut.jsx"><a href="statisticsDoughnut.jsx.html">statisticsDoughnut.jsx</a></td>
	<td data-value="16.67" class="pic low"><div class="chart"><div class="cover-fill" style="width: 16%;"></div><div class="cover-empty" style="width:84%;"></div></div></td>
	<td data-value="16.67" class="pct low">16.67%</td>
	<td data-value="6" class="abs low">1/6</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="6" class="abs low">0/6</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="20" class="pct low">20%</td>
	<td data-value="5" class="abs low">1/5</td>
	</tr>

<tr>
	<td class="file high" data-value="toggleCommentButton.jsx"><a href="toggleCommentButton.jsx.html">toggleCommentButton.jsx</a></td>
	<td data-value="100" class="pic high"><div class="chart"><div class="cover-fill cover-full" style="width: 100%;"></div><div class="cover-empty" style="width:0%;"></div></div></td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	</tr>

<tr>
	<td class="file low" data-value="topPostForm.jsx"><a href="topPostForm.jsx.html">topPostForm.jsx</a></td>
	<td data-value="7.41" class="pic low"><div class="chart"><div class="cover-fill" style="width: 7%;"></div><div class="cover-empty" style="width:93%;"></div></div></td>
	<td data-value="7.41" class="pct low">7.41%</td>
	<td data-value="54" class="abs low">4/54</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="41" class="abs low">0/41</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="19" class="abs low">0/19</td>
	<td data-value="7.55" class="pct low">7.55%</td>
	<td data-value="53" class="abs low">4/53</td>
	</tr>

<tr>
	<td class="file low" data-value="topPostFormContainer.jsx"><a href="topPostFormContainer.jsx.html">topPostFormContainer.jsx</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="29" class="abs low">0/29</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="43" class="abs low">0/43</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="11" class="abs low">0/11</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="29" class="abs low">0/29</td>
	</tr>

<tr>
	<td class="file low" data-value="whatYouNeedToKnow.jsx"><a href="whatYouNeedToKnow.jsx.html">whatYouNeedToKnow.jsx</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Oct 26 2018 15:51:30 GMT+0200 (CEST)
</div>
</div>
<script src="../../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../../sorter.js"></script>
</body>
</html>
