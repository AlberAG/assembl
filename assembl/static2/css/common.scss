body {
  background-color: $page-bkg-light;
  font-family: $font-family;
  font-size: $font-size-s;
  overflow-y: scroll;
}

section {
  margin: 30px 0;
}

.app-content {
  margin: 80px 0 0;
}

.max-container {
  margin: 0 auto;
  max-width: $app-max-width;
}

.background-grey {
  background-color: $page-bkg-grey;
}

.background-dark {
  background-color: $page-bkg-dark;
}

.inline {
  display: inline-block;
}

.left {
  float: left;
}

.right {
  float: right;
}

.center {
  text-align: center;
}

.border-right {
  border-right: $box-border;
}

.border-left {
  border-left: $box-border;
}

.padding {
  padding: 20px 0;
}

.margin-xs {
  margin-top: 5px;
}

.margin-s {
  margin-top: 10px;
}

.margin-m {
  margin-top: 20px;
}

.margin-l {
  margin-top: 40px;
}

.margin-xl {
  margin-top: 60px;
}

.margin-xxl {
  margin-bottom: 100px;
  margin-top: 100px;
}

.underline {
  border-bottom: $box-border;
  margin: 0 0 20px;
}

.hidden {
  display: none;
}

.shown {
  display: block;
}

.code {
  color: $grey3;
  display: inline-block;
}

#pDebug {
  display: none;
}