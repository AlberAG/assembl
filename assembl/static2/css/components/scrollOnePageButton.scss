.scroll-one-page {
  display: inline-block;
  position: fixed;
  cursor: pointer;
  z-index: 99;
  left: 50%;
  margin-left: -50px;

  span {
    border-radius: 50px 50px 0 0;
    height: 50px;
    width: 100px;
    display: block;
    background-color: $grey2;
    color: $white;
    text-align: center;

    &:hover {
        background-color: $grey3;
    }

    .icon {
      padding-top: 20px;
      padding-right: 1em;
      margin-bottom: 2px;
    }
  }
}