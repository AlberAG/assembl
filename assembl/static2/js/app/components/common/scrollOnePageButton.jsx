import React from 'react';

const scrollOnePageDown = () => {
  window.scrollBy(0, window.innerHeight);
};

const ScrollOnePageButton = ({ hidden }) => {
  const scrollOnePageTopPosition = window.innerHeight - 50;
  return (
    <a
      className={`scroll-one-page ${hidden ? 'hidden' : ''}`}
      onClick={scrollOnePageDown}
      style={{ top: scrollOnePageTopPosition }}
    >
      <span>
        <span className="icon assembl-icon-down-open">&nbsp;</span>
      </span>
    </a>
  );
};

export default ScrollOnePageButton;