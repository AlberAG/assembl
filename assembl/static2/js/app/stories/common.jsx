/*
  Stories for components/common
*/
import React from 'react';
import { storiesOf } from '@storybook/react';
import { Link } from 'react-router';

import Card from '../components/common/card';
import Loader from '../components/common/loader';
import SwitchButton from '../components/common/switchButton';

const cardContent = (
  <Link className="content-box">
    <div className="title-container center">
      <h3 className="light-title-3">Sunt quas ratione culpa</h3>
      <h4 className="light-title-4">
        Commodi architecto fugiat maxime quod. Dignissimos aut atque. Qui autem qui et sed voluptas iure suscipit. Est sit alias
        voluptatibus. Aliquam aperiam dolores dolores sint.
      </h4>
    </div>
  </Link>
);

const imgUrl = 'https://images.unsplash.com/uploads/141155339325423394b24/03982423';

storiesOf('Card', module)
  .add('default', () => <Card>{cardContent}</Card>)
  .add('with className (margin)', () => <Card className="margin-xxl">{cardContent}</Card>)
  .add('with imgUrl', () => <Card imgUrl={imgUrl}>{cardContent}</Card>);

storiesOf('Loader', module)
  .add('with text', () => <Loader textHidden={false} />)
  .add('with text and color', () => <Loader color="#891" />)
  .add('with text hidden', () => <Loader textHidden />)
  .add('with text hidden and color', () => <Loader color="#933" textHidden />);

storiesOf('SwitchButton', module)
  .add('default', () => <SwitchButton />)
  .add('checked by default', () => <SwitchButton defaultChecked />)
  .add('with label', () => <SwitchButton label="Please toggle me" />);