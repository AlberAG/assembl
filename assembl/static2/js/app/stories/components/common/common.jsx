/*
  Stories for components/common
*/
import React from 'react';
import { storiesOf } from '@storybook/react';
import { Link } from 'react-router';

import Card from '../../../components/common/card';
import Loader from '../../../components/common/loader';
import SwitchButton from '../../../components/common/switchButton';
import TabbedContent from '../../../components/common/tabbedContent';

const cardContent = (
  <Link className="content-box">
    <div className="title-container center">
      <h3 className="light-title-3">Sunt quas ratione culpa</h3>
      <h4 className="light-title-4">
        Commodi architecto fugiat maxime quod. Dignissimos aut atque. Qui autem qui et sed voluptas iure suscipit. Est sit alias
        voluptatibus. Aliquam aperiam dolores dolores sint.
      </h4>
    </div>
  </Link>
);

const imgUrl = 'https://images.unsplash.com/uploads/141155339325423394b24/03982423';

storiesOf('Card', module)
  .add('default', () => <Card>{cardContent}</Card>)
  .add('with className (margin)', () => <Card className="margin-xxl">{cardContent}</Card>)
  .add('with imgUrl', () => <Card imgUrl={imgUrl}>{cardContent}</Card>);

storiesOf('Loader', module)
  .add('with text', () => <Loader textHidden={false} />)
  .add('with text and color', () => <Loader color="#891" />)
  .add('with text hidden', () => <Loader textHidden />)
  .add('with text hidden and color', () => <Loader color="#933" textHidden />);

storiesOf('SwitchButton', module)
  .add('default', () => <SwitchButton />)
  .add('checked by default', () => <SwitchButton defaultChecked />)
  .add('with label', () => <SwitchButton label="Please toggle me" />);

const tabs = [
  {
    id: 'tab1',
    title: 'One',
    text: 'If we override the transmitter, we can get to the PNG bus through the solid state RSS alarm!'
  },
  { id: 'tab2', title: 'Two', text: 'I\'ll input the auxiliary RSS monitor, that should feed the ADP monitor!' },
  {
    id: 'tab3',
    title: 'Three',
    text: 'The ADP matrix is down, quantify the neural protocol so we can program the COM firewall!'
  }
];
const renderBody = (tab, idx) => (
  <div className="box">
    <h2>
      {tab.title} ({idx + 1})
    </h2>
    <p>{tab.text}</p>
  </div>
);

storiesOf('TabbedContent', module)
  .add('default', () => <TabbedContent tabs={tabs} renderBody={renderBody} />)
  .add('with divClassName', () => <TabbedContent divClassName="right" tabs={tabs} renderBody={renderBody} />)
  .add('with bodyRowClassName', () => <TabbedContent bodyRowClassName="right" tabs={tabs} renderBody={renderBody} />);