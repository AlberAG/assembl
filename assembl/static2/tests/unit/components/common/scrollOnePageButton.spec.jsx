import React from 'react';
import renderer from 'react-test-renderer';

import { ScrollOnePageButton } from '../../../../js/app/components/common/scrollOnePageButton';

describe('ScrollOnePageButton component', () => {
  it('should render a button to scroll one page down', () => {
    const component = renderer.create(<ScrollOnePageButton screenHeight={768} hidden={false} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render a button to scroll one page down that is hidden', () => {
    const component = renderer.create(<ScrollOnePageButton screenHeight={768} hidden />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});