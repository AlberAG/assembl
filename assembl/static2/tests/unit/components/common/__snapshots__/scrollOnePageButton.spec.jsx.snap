// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollOnePageButton component should render a button to scroll one page down 1`] = `
<a
  className="scroll-one-page "
  onClick={[Function]}
  style={
    Object {
      "top": 718,
    }
  }
>
  <span>
    <span
      className="icon assembl-icon-down-open"
    >
       
    </span>
  </span>
</a>
`;

exports[`ScrollOnePageButton component should render a button to scroll one page down that is hidden 1`] = `
<a
  className="scroll-one-page hidden"
  onClick={[Function]}
  style={
    Object {
      "top": 718,
    }
  }
>
  <span>
    <span
      className="icon assembl-icon-down-open"
    >
       
    </span>
  </span>
</a>
`;
